### DISCUSSION TITLE
Quest Pro Facial & Eye tracking branch

### DISCUSSION THREAD

**Discussion by korejan** [korejan/ALVR#62] [Announcements]
For those who aren't aware, there is a Quest Pro [facial-eye-tracking](https://github.com/korejan/ALVR/tree/facial-eye-tracking) ALXR branch and builds are available for this in the [ALXR-experimental](https://github.com/korejan/ALXR-experimental/releases) repository.

This version of ALXR supports both the Quest build for ALVR users and Windows build for (Air)Link users without the ALVR server. The latter works as a pure tool for (Air)link. Apps that currently support receiving the F&E data are VRChat via [VRCFT](https://github.com/Adjerry91/VRCFaceTracking) and work-in progress plugin for NeosVR.

More details/instructions about using VRCFT can be found [here](https://docs.google.com/document/d/1MFGxIrdh7U2tq368X_UthryceIsapwz6C7hydmnaWQM) and on the [VRCFT discord server](https://discord.gg/vrcft)

**Comment by Jia-WeiFang** [comment]
Dear @korejan,

If I want to use Oculus Quest Pro for the eye-tracking function, 
how to use the code in the **alvr/openxr-client/alxr-engine-sys/cpp/ALVR-OpenXR-Engine/src/external/oculus/XRSamples/XrEyeTrackingSocialFB/main.cpp** to get the gaze position?

I saw some codes that mention the eye-tracking part, should I merge them in alxr_engine.cpp or anywhere else?
Or it is already usable without editing anything?

Thanks a lot for updating the this!

**Reply by korejan** [reply]
@Jia-WeiFang Hi, the Quest pro facial & eye tracking is currently in branches of both in the [alvr repo](https://github.com/korejan/ALVR/tree/facial-eye-tracking) and [alxr-engine submodule](https://github.com/korejan/ALVR-OpenXR-Engine/tree/facial-eye-tracking). The file that you mentioned is sample code that comes with the Oculus OpenXR mobile sdk.

**Reply by Jia-WeiFang** [reply]
Dear @korejan,

Thank you for updating these features for Oculus Quest Pro!

May I wonder which parts of the code you add for the eye-tracking updates in this branch? (like there maybe some comments that refer to the new eye-tracking parts?) 
Since I have already implemented some functions in another branch, it will be efficient for me to move your eye-tracking code to my branch! (I'm afraid that I will miss something to move)

May I also wonder if I want to get the gaze position in runtime, how can I implement this function?
Should I directly add the sample code from the Oculus OpenXR mobile SDK to openxr_program.cpp?
Or is there anything else I should do?

Thanks a lot for your kind help!!!

Jia-Wei

**Reply by korejan** [reply]
@Jia-WeiFang The Quest Pro has a vendor specific extension for eye tracking that is a bit different from the multi vendor `XR_EXT_gaze_interaction` extension, the main difference being is the FB/Meta extension has two eye gaze poses. Yes you can check `openxr_program.cpp` in the [facial-eye-tracking](https://github.com/korejan/ALVR-OpenXR-Engine/tree/facial-eye-tracking) branch. Also try to find me on discord.
